engine/runtime-instrument-common/src/main/scala/org/enso/interpreter/instrument/job/ProgramExecutionSupport.scala

```diff
@@ -351,31 +353,45 @@ object ProgramExecutionSupport {
               ErrorResolver.getStackTrace(panic).flatMap(_.expressionId)
             )
         case _ =>
-          if (WarningsLibrary.getUncached.hasWarnings(value.getValue)) {
-            val warnings =
-              WarningsLibrary.getUncached.getWarnings(value.getValue, null)
-            val warningsCount = warnings.length
-            val warning =
-              if (warningsCount == 1) {
-                Option(
-                  ctx.executionService.toDisplayString(warnings(0).getValue)
+          val warnings =
+            Option.when(
```

Comment:
Replacing `if` with `Option.when`! I can only say: I am not fan of such aggressive _functional rewrites_.